### PR TITLE
[docs] Add description about enable-subscription-types

### PR DESCRIPTION
**Pull request by sijia-w**
Add descriptions for https://github.com/apache/pulsar/pull/9401.

### CODE REVIEW

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -340,26 +340,28 @@ subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions fr
...
 |haProxyProtocolEnabled | Enable or disable the [HAProxy](http://www.haproxy.org/) protocol. |false|
 | maxTopicsPerNamespace | The maximum number of persistent topics that can be created in the namespace. When the number of topics reaches this threshold, the broker rejects the request of creating a new topic, including the auto-created topics by the producer or consumer, until the number of connected consumers decreases. The default value 0 disables the check. | 0 |
+|getSubscriptionTypesEnabled| Enable all subscription types, that is exclusive, shared, failover and key_shared, for a namespace or a topic. | true |
+|setSubscriptionTypesEnabled| Enable one or more subscrption types for a namespace or a topic. Multiple subscription types are available, separated by `,`. You can set this parameter by `pulsar-admin namespaces set-subscription-types-enabled options` or `pulsar-admin topics set-subscription-types-enabled options` command. | null |
```
Comment (codelipenghui) [inline site2/docs/reference-configuration.md:352]: We don't have these 2 configurations in the broker.conf, the allowed config is `subscriptionTypesEnabled`. Here is the description in the broker.conf

```
# Enable subscription types (default is all type enabled)
# SubscriptionTypes : Exclusive,Shared,Failover,Key_Shared
# Example : Exclusive,Shared
# Above example will disable Failover and Key_Shared subscription types
subscriptionTypesEnabled=Exclusive,Shared,Failover,Key_Shared
```

**Review by codelipenghui — CHANGES_REQUESTED** [review]

File: site2/docs/reference-configuration.md (hunk 1 above)
Comment (sijia-w) [reply site2/docs/reference-configuration.md]: @codelipenghui Thank you for your comments. Can you please help review the updated description?

File: site2/docs/reference-configuration.md [hunk 2]
```diff
@@ -340,26 +340,27 @@ subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions fr
...
 | preciseTopicPublishRateLimiterEnable | Enable precise topic publish rate limiting. | false |
 | lazyCursorRecovery | Whether to recover cursors lazily when trying to recover a managed ledger backing a persistent topic. It can improve write availability of topics. The caveat is now when recovered ledger is ready to write we're not sure if all old consumers' last mark delete position(ack position) can be recovered or not. So user can make the trade off or have custom logic in application to checkpoint consumer state.| false |  
 |haProxyProtocolEnabled | Enable or disable the [HAProxy](http://www.haproxy.org/) protocol. |false|
 | maxTopicsPerNamespace | The maximum number of persistent topics that can be created in the namespace. When the number of topics reaches this threshold, the broker rejects the request of creating a new topic, including the auto-created topics by the producer or consumer, until the number of connected consumers decreases. The default value 0 disables the check. | 0 |
+|subscriptionTypesEnabled| Enable all subscription types, that is exclusive, shared, failover and key_shared, for a namespace. | true |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:351]:
```suggestion
|subscriptionTypesEnabled| Enable all subscription types for a namespace. Available subscription types are exclusive, shared, failover, and key_shared,  | true |
```
# Enable subscription types (default is all type enabled)
# SubscriptionTypes : Exclusive,Shared,Failover,Key_Shared
# Example : Exclusive,Shared
# Above example will disable Failover and Key_Shared subscription types
subscriptionTypesEnabled=Exclusive,Shared,Failover,Key_Shared

**Comment by codelipenghui** [comment]
@sijia-w @Anonymitaet `subscriptionTypesEnabled` is a broker-level configuration, not for a specific namespace or topic, it will affect the behavior of a broker. So we'd better remove `for a namespace, topic` from the description.

**Comment by Anonymitaet** [comment]
@codelipenghui thanks for your explanations, I've updated my comment accordingly.

File: site2/docs/reference-configuration.md [hunk 3]
```diff
@@ -340,26 +340,27 @@ subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions fr
...
-|maxMessagePublishBufferSizeInMB|The maximum memory size for broker handling messages sent from producers. If the processing message size exceeds this value, broker stops reading data from the connection. The processing messages means messages are sent to broker but broker have not sent response to the client. Usually the message are waiting to be written to bookies. It's shared across all the topics running in the same broker. The value `-1` disables the memory limitation. By default, it is 50% of direct memory.|N/A|
+|maxMessagePublishBufferSizeInMB|The maximum memory size for a broker to handle messages that are sent by producers. If the processing message size exceeds this value, the broker stops reading data from the connection. The processing messages refer to the messages that are sent to the broker but the broker has not sent response to the client. Usually the messages are waiting to be written to bookies. It's shared across all the topics running in the same broker. The value `-1` disables the memory limitation. By default, it is 50% of direct memory.|N/A|
```
Comment (Huanli-Meng) [inline site2/docs/reference-configuration.md:343]:
```suggestion
|maxMessagePublishBufferSizeInMB|The maximum memory size for a broker to handle messages that are sent by producers. If the processing message size exceeds this value, the broker stops reading data from the connection. The processing messages refer to the messages that are sent to the broker but the broker has not sent response to the client. Usually the messages are waiting to be written to bookies. It is shared across all the topics running in the same broker. The value `-1` disables the memory limitation. By default, it is 50% of direct memory.|N/A|
```

**Review by Huanli-Meng — COMMENTED** [review]
leave a minor comment. 

And one more question, whether this update is available for Release 2.7.1?

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/reference-configuration.md [hunk 4]
```diff
@@ -340,26 +340,27 @@ subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions fr
...
 | preciseTopicPublishRateLimiterEnable | Enable precise topic publish rate limiting. | false |
 | lazyCursorRecovery | Whether to recover cursors lazily when trying to recover a managed ledger backing a persistent topic. It can improve write availability of topics. The caveat is now when recovered ledger is ready to write we're not sure if all old consumers' last mark delete position(ack position) can be recovered or not. So user can make the trade off or have custom logic in application to checkpoint consumer state.| false |  
 |haProxyProtocolEnabled | Enable or disable the [HAProxy](http://www.haproxy.org/) protocol. |false|
 | maxTopicsPerNamespace | The maximum number of persistent topics that can be created in the namespace. When the number of topics reaches this threshold, the broker rejects the request of creating a new topic, including the auto-created topics by the producer or consumer, until the number of connected consumers decreases. The default value 0 disables the check. | 0 |
+|subscriptionTypesEnabled| Enable all subscription types. Available subscription types are exclusive, shared, failover, and key_shared.  | true |
```
Comment (codelipenghui) [inline site2/docs/reference-configuration.md:351]:
```suggestion
|subscriptionTypesEnabled| Enable all subscription types. Available subscription types are exclusive, shared, failover, and key_shared.  | Exclusive,Shared,Failover,Key_Shared |
```

**Review by codelipenghui — CHANGES_REQUESTED** [review]

**Comment by Anonymitaet** [comment]
@Huanli-Meng sorry, I saw your comment just now. Confirmed w/ @congbobo184, this update is not available for 2.7.1.

**Comment by Huanli-Meng** [comment]
> @Huanli-Meng sorry, I saw your comment just now. Confirmed w/ @congbobo184, this update is not available for 2.7.1.

It's OK. Thanks for your feedback

**Comment by Anonymitaet** [comment]
@codelipenghui your comments have been incorporated, could u PTAL?

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**